packages/preview/outline-summaryst/0.1.0/src/outline-summaryst.typ

```diff
@@ -0,0 +1,123 @@
+#import "helpers.typ": calc-elem-size, make-title, make-foreword
+
+// randomly generated IDs
+#let gs = state("VIvouA", ())
+#let elemcounter = counter("k1lRPA")
+
+#let supl = [Custom Heading]
+
+
+#let outline-summaryst(
```

Comment:
Is this function needed at all? Wouldn't it make more sense to provide a function that generates just the fancy outline instead? 

Since your headings use a special `make-heading` function instead of the syntax, the heading show rule could also be included with that. That way it would be even more composable.

Reply:
I've changed the behavior of the both functions to be more composable based on your feedback (note that I renamed `outline-summaryst` to `style-outline`).

To answer your question of why `outline-summaryst` is not a function that generates the fancy outline, this is because some templates include an `#outline()` by default, so this package could not be used alongside such templates. The current behavior, where we have a show rule for `outline.entry`, solves this issue